indexer/src/main/java/bio/terra/tanagra/indexing/job/dataflow/WriteRollupCounts.java

```diff
@@ -159,9 +160,29 @@ public void run(boolean isDryRun) {
       LOGGER.info("Rollup counts source SQL is defined. Skipping Dataflow job.");
     }
 
-    // Dataflow jobs can only write new rows to BigQuery, so in this second step, copy over the
-    // count values to the corresponding column in the index entity main table.
-    copyFieldsToEntityTable(isDryRun);
+    final int maxRetries = 5;
+    var attempt = 0;
+
+    while (attempt < maxRetries) {
+      try {
+        // Dataflow jobs can only write new rows to BigQuery, so in this second step, copy over the
+        // count values to the corresponding column in the index entity main table.
+        copyFieldsToEntityTable(isDryRun);
+        return; // Exit if successful
+      } catch (BigQueryException e) {
+        if (e.getMessage().contains("due to concurrent update")) {
+          attempt++;
+          LOGGER.info("Attempt {} failed: {}", attempt, e.getMessage());
+
+          if (attempt == maxRetries) {
+            LOGGER.info("Max retries reached. Giving up.");
```

Comment:
minor: would a sleep help?

Reply:
It's working in my test runs but won't hurt to add it.